### PR TITLE
`weather-mv bq` can now load Zarr into BigQuery.

### PR DESCRIPTION
**Pull request by alxmrs**
This CL adds a Zarr reading option to `open_dataset` and to the BigQuery data sink. In addition, we provide some small refactoring to the BigQuery transform to make it more succinct.

### CODE REVIEW

**Comment by alxmrs** [comment]
CC: @chaitanya176

File: weather_mv/loader_pipeline/bq.py [hunk 1]
```diff
@@ -180,26 +185,72 @@ def __post_init__(self):
...
+                    if c not in it and (not self.variables or c in self.variables):
+                        row[c] = to_json_serializable_type(ensure_us_time_resolution(row_ds[c].values))
+
+                # Add import metadata.
+                row[DATA_IMPORT_TIME_COLUMN] = self.import_time
+                row[DATA_URI_COLUMN] = uri
+                row[DATA_FIRST_STEP] = first_time_step
+                row[GEO_POINT_COLUMN] = fetch_geo_point(row['latitude'], row['longitude'])
+
+                # 'row' ends up looking like:
```
Comment (chaitanya176) [inline weather_mv/loader_pipeline/bq.py:241]: @alxmrs is this bq schema planned for Zarr?

File: weather_mv/loader_pipeline/bq.py [hunk 2]
```diff
@@ -151,10 +149,17 @@ def validate_arguments(cls, known_args: argparse.Namespace, pipeline_args: t.Lis
                             region=pipeline_options_dict.get('region'))
             logger.info('Region validation completed successfully.')
 
+        if known_args.zarr:
+            # Encourage use of correct output_path format.
+            _, out_ext = os.path.splitext(known_args.output_path)
```
Comment (chaitanya176) [inline weather_mv/loader_pipeline/bq.py:154]: We may need to define a zarr output format for namespace kwargs

Comment (alxmrs) [reply weather_mv/loader_pipeline/bq.py:154]: I don't quite understand. What are namespace kwargs?

Comment (chaitanya176) [reply weather_mv/loader_pipeline/bq.py:154]: Hey Alex, I was referring to the error in the build: AttributeError: 'Namespace' object has no attribute 'output_path'. Seems like we may have to define it.

**Comment by mahrsee1997** [comment]
Not required anymore as the follow up PR #333 merged into main.

**Comment by mahrsee1997** [comment]
Not required anymore as the follow up PR https://github.com/google/weather-tools/pull/333 merged into main.

**Pull request closed without merging**